src/main/java/de/tum/in/www1/artemis/service/GroupNotificationService.java

```diff
@@ -93,6 +93,28 @@ public void notifyGroupAboutNewAnswer(StudentQuestionAnswer studentQuestionAnswe
         notifyGroupAboutExercise(studentQuestionAnswer.getQuestion().getExercise(), title, notificationText, "newAnswer", GroupNotificationType.INSTRUCTOR);
     }
 
+    public void notifyStudentGroupAboutAttachmentChange(Attachment attachment, String notificationText) {
+        if (attachment.getReleaseDate() != null && !attachment.getReleaseDate().isBefore(ZonedDateTime.now())) {
+            return;
+        }
+        GroupNotification groupNotification = new GroupNotification();
+        groupNotification.setCourse(attachment.getLecture().getCourse());
+        groupNotification.setType(GroupNotificationType.STUDENT);
+        groupNotification.setNotificationDate(ZonedDateTime.now());
+        groupNotification.setTitle("Attachment " + attachment.getName() + " updated");
+        groupNotification.setText(notificationText);
+        JsonObject target = new JsonObject();
+        target.addProperty("message", "attachmentUpdated");
+        target.addProperty("id", attachment.getLecture().getId());
+        target.addProperty("entity", "lectures");
+        target.addProperty("course", attachment.getLecture().getCourse().getId());
+        target.addProperty("mainPage", "overview");
+        groupNotification.setTarget(target.toString());
+        groupNotification.setAuthor(userService.getUser());
+        String topic = "/topic/course/" + groupNotification.getCourse().getId() + "/" + groupNotification.getType();
```

Comment:
I feel like composing the topic is something that should not be necessary here.
Can this be done in the `saveAndSendGroupNotification(topic, groupNotification)` method?
Or can this even be refactored to be `groupNotification.getTopic()`?

In general, as a developer using the Notification API, I do not want to care about the underlying web sockets. If this where to switch to rediscover, mqtt or whatever is hipp tmr, this should not impact the API of the Notification Component.

Reply:
Valid point, I suggest that we postpone this comment to another pull request, in which we refactor the API for notifications. I will create a JIRA ticket for this. Then we can merge this pull request, because we need the functionality of lectures and attachments urgently :)